app/src/main/java/net/christianbeier/droidvnc_ng/Defaults.kt

```diff
@@ -109,6 +110,40 @@ class Defaults {
          */
         this.scaling = 1.0f / Utils.getDisplayMetrics(context, Display.DEFAULT_DISPLAY).density.coerceAtLeast(1.0f)
 
+        /*
+            Try read defaults from app restrictions
+         */
+
+        val restrictionsManager =
+            context.getSystemService(Context.RESTRICTIONS_SERVICE) as android.content.RestrictionsManager
+        val appConfig = restrictionsManager.applicationRestrictions
+        if (appConfig != null && appConfig.size() > 0) {
+            this.port = appConfig.getInt("port", this.port)
+            this.portReverse = appConfig.getInt("portReverse", this.portReverse)
+            this.portRepeater = appConfig.getInt("portRepeater", this.portRepeater)
+            this.fileTransfer = appConfig.getBoolean("fileTransfer", this.fileTransfer)
+            this.viewOnly = appConfig.getBoolean("viewOnly", this.viewOnly)
+            this.showPointers = appConfig.getBoolean("showPointers", this.showPointers)
+            this.password = appConfig.getString("password", this.password) ?: this.password
+            this.startOnBoot = appConfig.getBoolean("startOnBoot", this.startOnBoot)
+            this.startOnBootDelay = appConfig.getInt("startOnBootDelay", this.startOnBootDelay)
+
+            val scalingStr = appConfig.getString("scaling", "0.0")
+            try {
+                val scaling = scalingStr.toFloat()
+                if (scaling > 0.0f)
+                    this.scaling = scaling
+            } catch (e: NumberFormatException) {
+                Log.w(TAG, "Invalid scaling value in app restrictions: $scalingStr")
+            }
+
+            val accessKey = appConfig.getString("accessKey", "")
+            if (accessKey != null && accessKey.isNotEmpty())
+                this.accessKey = accessKey
+
```

Comment:
pls add a Log.i call indicating that defaults were loaded from this source and any JSON ignored